nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/model/DocumentModelResolver.java

```diff
@@ -221,15 +222,15 @@ protected void resolve(Object value, Object context, BiConsumer<CoreSession, Doc
                         session = (CoreSession) context;
                     } else {
                         // otherwise open a new one
-                        closeableCoreSession = CoreInstance.openCoreSession(ref.repo);
+                        closeableCoreSession = Framework.doPrivileged(() -> CoreInstance.openCoreSession(ref.repo));
                         session = closeableCoreSession;
                     }
                 } else {
                     // use session from context
                     session = (CoreSession) context;
                     if (session == null) {
                         // use the default repository if none is provided in the context
-                        closeableCoreSession = CoreInstance.openCoreSession(null);
+                        closeableCoreSession = Framework.doPrivileged(() -> CoreInstance.openCoreSession(null));
```

Comment:
If I understand correctly it means that if you set the value of a docid property to the id of the document you don't have access to (through various REST APIs), then you can get the document itself through the resolver dereference.
So to me that looks like a serious security issue.
👎